### DISCUSSION TITLE
关于saiblo的黑白棋的计时以及超时界定

### DISCUSSION THREAD

**Discussion by jiqingsong** [physics-data/faq#62] [Q&A] [answered]
请问saiblo平台的计时标准是什么？
情况描述：我在使用time.time(),time.process_time()和time.perf_counter()计时时，1、3结果相同，2结果与其余二者不同。
其中，若要求time.time()或time.perf_counter在3秒时结束程序，则在saiblo上显示的ai运算时间只有0.1秒左右。
若要求time.process_time()在3秒时结束程序，则saiblo会在0.6秒左右宣布程序超时。

**Comment by Harry-Chen** [comment]
`time` 和 `perf_counter` 是不正确的，这样测量的是 wall time。考虑到多进程调度的影响，你的程序实际获得的 CPU 时间肯定低于这个数。

`process_time` 看起来是一个比较保守的计时方法，但程序第一次调用时可能还有其他开销（如 `import` 开销），因此不应该只看单次搜索的时间。

**Reply by jiqingsong** [reply]
所以saiblo的超时计时是以cpu时间来界定嘛？
另见下回答中的对局编号与图片，这是用process_time计时的结果，无论怎么说，其他开销占用2s+或多或少有点夸张，或者说这个额外开销有其他方面嘛（import开销我尝试过，将部分import注释后重跑，结果仍然超时，import看起来只占用了最多不到100ms的时间）

**Comment by rcy17** [comment]
“saiblo会在0.6秒左右宣布程序超时”有对局编号吗？

**Reply by jiqingsong** [reply]
2410015
本局190ms超时
![image](https://user-images.githubusercontent.com/37206109/130537964-7a52af33-fa03-4aa4-8eb2-528e5af2df78.png)
其中，我在调试时输出了当前process_time()与开始时process_time()之差
![image](https://user-images.githubusercontent.com/37206109/130538047-402287f4-d1f9-40da-a039-016d29b5b279.png)
看到确实只跑了200ms左右

**Reply by jiqingsong** [reply]
程序进入死循环后会提前被判定超时吗？

**Comment by jiqingsong** [comment]
去掉多余的ask_next_board后解决

**Comment by rcy17** [comment, marked as answer]
计时显示的是用户态时间，但判定超时既看用户态时间也看wall time，wall time相比用户态时间留有500ms的余地，而频繁调用ask_next_board 会有较多的系统调用（主要是通信等待）时间。请避免在自己程序里引入较多的系统调用，包括 print

**Reply by jiqingsong** [reply]
谢谢学长
